common/queue/src/main/java/org/thingsboard/server/queue/usagestats/DefaultTbApiUsageClient.java

```diff
@@ -47,8 +51,9 @@
     @Value("${usage.stats.report.interval:10}")
     private int interval;
 
-    @SuppressWarnings("unchecked")
-    private final ConcurrentMap<TenantId, AtomicLong>[] values = new ConcurrentMap[ApiUsageRecordKey.values().length];
+    private final EnumMap<ApiUsageRecordKey, ConcurrentMap<EntityId, AtomicLong>> stats = new EnumMap<>(ApiUsageRecordKey.class);
```

Comment:
Please change this to something like:

private final EnumMap<ApiUsageRecordKey, ConcurrentMap<OwnerIdKey, AtomicLong>> stats = new EnumMap<>(ApiUsageRecordKey.class);

where OwnerIdKey will be TenantId + EntityId.. this will help to remove redundant tenants map..